sunpy/coordinates/frames.py

```diff
@@ -449,7 +449,28 @@ class Helioprojective(SunPyBaseCoordinateFrame):
 
     rsun = Attribute(default=_RSUN.to(u.km))
     observer = ObserverCoordinateAttribute(HeliographicStonyhurst)
-
+    
+    
+    def angular_radius(t='now'):
+  
+        return _angular_radius(constants.radius, earth_distance(t))
```

Comment:
```suggestion
        return _angular_radius(self.rsun, self.observer.radius)
```
The coordinate can define its own radius of the Sun, and the observer is not necessarily at Earth.